wasm/src/lib.rs

```diff
@@ -1,33 +1,133 @@
 mod wasm_builtins;
 
+extern crate js_sys;
+extern crate num_bigint;
 extern crate rustpython_vm;
 extern crate wasm_bindgen;
 extern crate web_sys;
 
-use rustpython_vm::VirtualMachine;
+use num_bigint::BigInt;
 use rustpython_vm::compile;
-use rustpython_vm::pyobject::AttributeProtocol;
+use rustpython_vm::pyobject::{self, IdProtocol, PyObjectRef, PyResult};
+use rustpython_vm::VirtualMachine;
 use wasm_bindgen::prelude::*;
 use web_sys::console;
 
+fn py_str_err(vm: &mut VirtualMachine, py_err: &PyObjectRef) -> String {
+    vm.to_pystr(&py_err)
+        .unwrap_or_else(|_| "Error, and error getting error message".into())
+}
+
+fn py_to_js(vm: &mut VirtualMachine, py_obj: &PyObjectRef) -> JsValue {
+    use pyobject::PyObjectKind;
+    let py_obj = py_obj.borrow();
+    match py_obj.kind {
```

Comment:
The `kind` member is a bit of a misnomer, it might be renamed in future, but it is more the content of a python object in rust terms. To check for types, it is better to test with the isinstance method.